Limit HistoricPercentile param to valid options

CloudNiner opened the pull request:
## Overview

Update component to use dropdown since we only calculate historic percentiles for a few select values. 


### Demo

![screen shot 2017-11-20 at 16 38 01](https://user-images.githubusercontent.com/1818302/33042833-361d0354-ce11-11e7-8405-94441cbb1e3c.png)


## Testing Instructions

See #15 for instructions on how to prepare your lab VM to test.

Then, load up any of the percentile+historic indicators, such as "Extreme Precip Events" and ensure that 1. a default of 95 is loaded and 2. that you are limited to the values specified in #14 

Closes #14

fungjj92 commented:
I just noticed that Extreme Cold Events doesn't have extra params in Lab because it was left out of `extra-params.constants.ts`. Not necessarily part of this work, but convenient to roll in here.

<img width="645" alt="screen shot 2017-11-21 at 3 10 35 pm" src="https://user-images.githubusercontent.com/10568752/33094435-302a3710-cece-11e7-8d9f-c1f0c22570d4.png">

fungjj92 commented on src/lib/modules/charts/components/extra-params/percentile-historic.component.ts:
```diff
@@ -48,7 +49,7 @@ export class PercentileHistoricComponent implements AfterViewInit, OnInit {
         this.percentileHistoricForm = this.formBuilder.group({
             historicCtl: [this.extraParams.historic_range || this.defaultHistoric],
-            percentileCtl: [this.extraParams.percentile || this.defaultPercentile, Validators.required]
+            percentileCtl: [this.extraParams.percentile || HistoricPercentileParam.NinetyFive]
```
Awk. The API has a different default for different extreme indicators. 1, 99, and 99. I'm not so concerned about matching _every_ case, but maybe let's do `NinetyNine`.

https://docs.climate.azavea.com/indicators.html#extreme-cold-events

CloudNiner replied:
Oh. Yeah that isn't great. I'll see what I can do to clean this up.

fungjj92 commented on src/lib/modules/api/models/historic-percentile-param.enum.ts:
```diff
@@ -0,0 +1,13 @@
+}
+
+const keys = Object.keys(HistoricPercentileParam).filter(k => {
+  return typeof HistoricPercentileParam[k as any] === 'number';
```
This stuck out to me as very robust haha. Curiosity, why did you write it like this?

CloudNiner replied:
Can't claim original authorship over that strategy, it [came from here](https://github.com/Microsoft/TypeScript/issues/17198#issuecomment-315400819) as a suggestion by one of the TypeScript devs (I think).

CloudNiner commented:
Updated. Added `extreme_cold_events` and set default dynamically based on indicator.

CloudNiner commented:
Updated and rebased against develop after #12 went in. 

There's now a nasty chart flash if the chart renders before the HistoricParamsService.list() returns the valid options which sets the default. I don't want to poison this PR with more changes, so I'll submit a separate PR to address that or make an issue once this is merged.
